pkg/kindsys/kindcat_composable.cue

```diff
@@ -15,15 +19,17 @@ Composable: S={
 	// schemaInterface is the name of the Grafana schema interface implemented by
 	// this Composable kind. The set is open to ensure forward compatibility of
 	// Grafana and tooling with any additional schema interfaces that may be added.
-//	schemaInterface: or([ for k, _ in schemaInterfaces {k}, string])
 	schemaInterface: string
+	// TODO is it worth doing something like below, given that we have to keep this set open for forward compatibility?
+//	schemaInterface: or([ for k, _ in schemaInterfaces {k}, string])
 
 	let schif = schemaInterfaces[S.schemaInterface]
 
 	// lineage is the Thema lineage containing all the schemas that have existed for this kind.
 	// The name of the lineage is constrained to the name of the schema interface being implemented.
-//	lineage: thema.#Lineage & {name: S.schemaInterface, joinSchema: schif.interface}
-	lineage: { joinSchema: (schif.interface | *{}) }
+// FIXME cuetsy currently gets confused by all the unifications - maybe openapi too. Do something like the following after thema separates joinSchema/constraint expression
+//	lineage: { joinSchema: schif.interface }
+//	lineage: { joinSchema: (schif.interface | *{}) }
```

Comment:
The fact that we're not injecting a `joinSchema` here basically means that we're not actually enforcing the schema interface-defined rules on the composable kinds declared by plugins. That means that `ParsePluginFS` validation won't fail if, for example, it encounters a `PanelCfg` lineage that lacks a `PanelOptions` field.

I've commented it out for now because there's a bug in how cuetsy deals with the additional unifications when it comes to generating enums. It's mostly just a sign of cuetsy needing a deeper refactor, but i believe the problem will likely also go away when grafana/thema#82 goes in. So, given that we're still not realistically rolling this out outside of Grafana core for at least a few more months and it's easy enough to notice and fix any plugin not following the pattern in core, this seems like an OK hack for now.

Reply:
no, but i think it would be really interesting to articulate that as a milestone/goal for the system overall - when we allow external plugin developers to join the party.

It's not something i've been directly thinking about much because it's always struck me as your squad's decision. But it's organically come up in conversation twice this week, which suggests the time has probably come to start working that out :)

This particular item does need a follow-up grafana-side.

Reply:
i've created #61838